soc/arm/arm/designstart/soc.h

```diff
@@ -9,4 +9,6 @@
 
 #include <devicetree.h>
 
+#define __MPU_PRESENT CONFIG_CPU_HAS_ARM_MPU
```

Comment:
Where do you define CONFIG_CPU_HAS_ARM_MPU for the Cortex-M3 SoC?

Reply:
It's defined in `boards/arm/arty/Kconfig.board` since the MPU is optional.

Reply:
That's why I could not find it 👍 

This is not ideal; CPU_HAS_ARM_MPU is always selected by SoC Kconfig. It is optional for the SoC; I wonder if you could reflect this in the SoC definition somehow.

Reply:
I could introduce a SoC specific Kconfig for it and have that select CPU_HAS_ARM_MPU. Would that be better?

Reply:
so, normally, it should be selected in: 
```
config SOC_ARM_DESIGNSTART_FPGA_CORTEX_M3
	bool "ARM Cortex-M3 DesignStart FPGA"
	select CPU_CORTEX_M3
	select XIP
	select HAS_SEGGER_RTT
```

but if you think that it is optional in this SoC, you can add a soc VARIANT and select it there. This is also done in several ARM SoC definitions.

Reply:
I modeled this similar as to how you recommended handling the optional SysTick is handled for the Cortex-M1. Are you sure CPU_HAS_ARM_MPU should be handled differently?

Reply:
No, it should not be handled differently. Just that it it starts getting weird with all these SoC-related options defined in Board level.

There's a good reason for this, though; this is an FPGA definition at the BOARD level. So we can live with this.

Reply:
Okay. I will leave it as-is, then.